### DISCUSSION TITLE
Frage zu Entity Names

### DISCUSSION THREAD

**Discussion by Knechtie** [SukramJ/aiohomematic#267] [Q&A]
Hallo,

erstmal vielen Dank für alles was ihr an dem Projekt in den letzten Monaten geleistet habt 🙏🏻.

Ich habe eine Frage zu der Namensgebung der Entitäten. Vor ein paar Wochen habe ich bei meinem Testsystem bemerkt, dass bei einer Neuinstallation die Namen der CCU-Kanäle übernommen werden. Daraufhin habe ich mein Produktivsystem auf hahomematic umgestellt. 
Bei einer Neuinstallation einer neueren Version von hahomematic werden nun allerdings wieder Namen wie "hahm_00139affffffff_4" generiert.
Mache ich irgendetwas falsch bei der Konfiguration (ich habe die Felder für XML & JSON RPC leer gelassen)? 
Ich glaube am Config-Flow hat sich zwischenzeitlich etwas geändert.

**Comment by Knechtie** [comment]
Liegt es am nicht angegebenen JSON-RPC Port? (Das habe ich gerade in anderen Diskussionen aufgeschnappt).
Ich glaube mich zu erinnern, dass der Port im alten Config-Flow nach dem ersten Klick auf "Absenden" eingetragen wurde, da bin ich mir aber nicht sicher ob das wirklich so war.
Man könnte sich überlegen, ob man entsprechende Standardwerte einträgt oder irgendwie erwähnt, z.B. im Namensfeld der Eingabebox.

**Reply by SukramJ** [reply]
Die [Einstellungen](https://github.com/danielperna84/custom_homematic#central) müssen nur geändert werden wenn Nicht-Standard Ports verwendet werden.
Wichtig sind auch die [Firewall](https://github.com/danielperna84/custom_homematic#firewall) Einstellungen.

Damit die Namen geladen werden können müssen in der CCU-Firewall die Ports 80 (443 TLS) geöffnet sein.
Damit die entity_ids auch mit richtigem Namen erstellt werden, solltest Du die Integration noch mal neu installieren.

**Reply by danielperna84** [reply]
Könnte man nicht einen Service anlegen mit dem man als User zur Laufzeit die Möglichkeit hat die Namen neu zu laden? Bzw., zumindest den Cache leeren, damit sie beim nächsten Neustart neu geladen werden. 🤔

**Reply by SukramJ** [reply]
Den `names cache` hab ich schon ausgebaut, der der keine (kaum) Vorteile gebracht hat, d.h. beim Reload der Integration oder dem Neustart von HA werden die Namen neu geladen.

Ansonsten [siehe](https://github.com/danielperna84/hahomematic/wiki/Howto's#rename-of-device-in-ccu-not-reflected-in-home-assistant).
